src/Admin/AbstractAdmin.php

```diff
@@ -2254,15 +2280,14 @@ public function addChild(AdminInterface $child)
 
         $this->children[$child->getCode()] = $child;
 
-        $child->setParent($this);
-
         // NEXT_MAJOR: remove $args and add $field parameter to this function on next Major
 
         $args = \func_get_args();
 
         if (isset($args[1])) {
```

Comment:
Should we also replace `isset()` with `array_key_exists()` or `func_num_args()` here?

Reply:
null is not a valid value, so isset seems enough.

Reply:
Where is it validated? The argument 2 is not defined in this method or its interface.

Reply:
It's validated in the `setParent` method, and only string are accepted.
So if `null` is passed to `addChild` it's better to not pass the value to `setParent`.